en/services/mission.md

```diff
@@ -379,6 +379,51 @@ Note:
+While executing, the vehicle will progress sequentially through the mission items, looping and jumping in response to jump commands.
+
+The mission can be paused by changing to another flight mode, such as Hold/loiter, and restarted by changing back to mission mode (when armed).
+On restart the vehicle will _generally_ continue towards the same waypoint item as when it was paused.
+In some cases flight stacks may start from the preceding waypoint, for example, a paused camera survey may restart on the previous waypoint to ensure the whole paused leg is captured.
+Note that there is no concept of "stopping" a mission independent of pausing it.
+
+On completion of all mission items the mission will "complete".
+At this point some flight stacks will loiter within the mission mode, while others will transition to some other mode (such as "Hold") and not allow you to return to mission mode.
+In either case, the mission cannot be restarted unless it is [reset](#resetting-missions) (which usually happens after the vehicle lands and disarms).
+
+While the mission is executing the current mission item can be changed via MAVLink ([MAV_CMD_DO_SET_MISSION_CURRENT](#MAV_CMD_DO_SET_MISSION_CURRENT)).
+Note however that this does not reset the jump counters, and if the mission is complete, will not restart the mission.
+
+### Starting Missions
+
+A flight stack will usually run mission checks before allowing a vehicle to enter the mission executing mode (typically checks are run when first executing a mission, but some flight stacks might also check before continuing a paused mission).
+
+Provided the checks pass, you can start a mission by either switching to mission mode and arming, or arming and switching to mission mode.
+With mavlink we can change the mode using [MAV_CMD_DO_SET_MODE](../messages/common.md#MAV_CMD_DO_SET_MODE) to set the mode and ARM using[MAV_CMD_COMPONENT_ARM_DISARM](../messages/common.md#MAV_CMD_COMPONENT_ARM_DISARM).
+
+If supported the [MAV_CMD_MISSION_START](../messages/common.md#MAV_CMD_MISSION_START) command will both arm the vehicle (if necessary) and switch to the mission mode.
+This command also allows you to specify a particular start and end item in the mission.
+
+### Pausing/Stopping Missions
+
+Pause and stop missions by changing mode.
+In MAVLink this can be done using [MAV_CMD_DO_SET_MODE](../messages/common.md#MAV_CMD_DO_SET_MODE) (generally a vehicle will also support changing modes via an RC controller/joystick too).
+
+### Resetting Missions
+
+Once a mission has started it will iterate through the mission items, jumping and looping as indicated by jump mission items, until it completes.
+While the mission is running you can change the current mission item, but this does not reset the jump counters used for loops, and once the mission is complete, changing the current mission item will not restart it.
+
+The mission will reset after you land and the vehicle disarms.
```

Comment:
Are you sure?

Reply:
I thought so. What did you think - only on reboot?

Reply:
On reboot for sure.

Reply:
And on re-upload, or manual reset to 0.

Reply:
ArduPlane will not reset counters if you land the vehicle in auto.  It's a bit more complicated because some missions have the vehicle land and disarm, but then be able to resume the mission from where they are in their sequence when they're armed in auto again.

Reply:
> And on re-upload, or manual reset to 0.

@julianoes Just to be clear, 
- I understand reupload on the ground will reset the counters and the current waypoint. 
- For upload in air the current waypoint does not change right? On PX4 you "finish the previously uploaded waypoint" and then the waypoint number increments and you progress to that waypoint in the newly uploaded mission.

> On manual reset to 0

What do you mean by this? Is this the same as ArduPilot - setting mission current to 0 resets the whole mission? If so, that would be good. 

> ArduPlane will not reset counters if you land the vehicle in auto. It's a bit more complicated because some missions have the vehicle land and disarm, but then be able to resume the mission from where they are in their sequence when they're armed in auto again.

@peterbarker Thanks very much. This caveat is true of all PX4 vehicle too - its a configuration setting or a mission type setting. I will add a comment.

Reply:
> For upload in air the current waypoint does not change right? On PX4 you "finish the previously uploaded waypoint" and then the waypoint number increments and you progress to that waypoint in the newly uploaded mission.

This is ground station dependent. I think QGC does that. But if you just upload the whole mission, it will go back to 0.

> setting mission current to 0 resets the whole mission? If so, that would be good.

I think so, except things like DO_JUMP counters.

Reply:
@julianoes 

> I think so, except things like DO_JUMP counters.

So it will remove the "complete" bit and restart the mission, but with the wrong counters? That's screwed up. Can we make it also reset the counters? (i.e. as part of this protocol) and make not resetting them non-standard.

FYI also @peterbarker my concern here is that we updated [MAV_CMD_DO_SET_MISSION_CURRENT](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_MISSION_CURRENT) to add a reset option because there was not way to reset the counters. Now it seems like setting the mission item to 0 would do this. That's a better approach. So we should standardize.
I.e. I think we'd be best off removing that separate parameter and stating that setting to zero should reset the counters and the complete bit and reset to zero. Thoughts?

Reply:
> So it will remove the "complete" bit and restart the mission, but with the wrong counters? That's screwed up. Can we make it also reset the counters? (i.e. as part of this protocol) and make not resetting them non-standard.

That's the way I made it in 2013 or so. It might have been fixed since. You could try it out...

Reply:
I doubt it. In any case, Peter commented somewhere here he prefers that counters are reset explicitly. So we probably need to implement that second parameter. I looked at it once but couldn't work out the right place to even add the command.